### PR TITLE
feat: display packed files list in `pnpm pack`

### LINKED ISSUE #8676: Display packed files list info in pnpm pack
### Contribution

- [X] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

### Describe the user story

It's clearer to display packed files list info when packing

### Describe the solution you'd like

The same as npm pack does

### Describe the drawbacks of your solution

_No response_

### Describe alternatives you've considered

_No response_

### PR DESCRIPTION
**Pull request by lvqq**
close #8676 

It's useful to display packed files since `pnpm pack` only shows tarball filename now

## Before
image

## After
image

### CODE REVIEW

File: releasing/plugin-commands-publishing/src/pack.ts [hunk 1]
```diff
@@ -111,6 +113,8 @@ export async function handler (
       [path.join(dir, 'package.json')]: publishManifest as Record<string, unknown>,
     },
   })
+  // display order by name
+  globalInfo(`${chalk.blueBright('Tarball Contents')}\n${files.sort().join('\n')}`)
```
Comment (zkochan) [inline releasing/plugin-commands-publishing/src/pack.ts]: Don’t send ANSI escape codes to the `@pnpm/logger`.

There are 2 possible ways to implement it:

* return the whole output from the handler function
* log the list of files as a debug message and render the output in `@pnpm/default-reporter`.

Comment (lvqq) [reply releasing/plugin-commands-publishing/src/pack.ts]: done

File: releasing/plugin-commands-publishing/src/pack.ts [hunk 2]
```diff
@@ -111,6 +112,8 @@ export async function handler (
       [path.join(dir, 'package.json')]: publishManifest as Record<string, unknown>,
     },
   })
+  // display order by name
+  const packFiles = files.sort().join('\n')
```
Comment (zkochan) [inline releasing/plugin-commands-publishing/src/pack.ts]: Let’s make sort more predictable:
```suggestion
  const packFiles = files.sort((a, b) => a.localeCompare(b, 'en')).join('\n')
```

Comment (zkochan) [reply releasing/plugin-commands-publishing/src/pack.ts]: Also, this line of code is not needed at this point. Only at the end of the function

**Review by zkochan — APPROVED** [review]

**Comment by karlhorky** [comment]
This ended up breaking CI for us, since we were relying on the stdout from `pnpm pack` only containing the single line with the path.

In case anyone else is getting failures because they were relying on the `pnpm pack` output, here's how we fixed it:

```diff
 import { execa } from 'execa';

 // ...

     // Pack and install Preflight globally
-    const { stdout: pnpmPackTarballPath } = await execa`pnpm pack`;
+    const pnpmPackTarballPath = (await execa`pnpm pack`).stdout
+      .split('\n')
+      .find((line) => line.match(/^upleveled-preflight-.*\.tgz$/));
+
+    if (!pnpmPackTarballPath) {
+      throw new Error('Failed to find the tarball path in `pnpm pack` output');
+    }
```

**Comment by zkochan** [comment]
Sorry for that. It actually broke some of my integrations as well. We should add a JSON output option.

**Comment by karlhorky** [comment]
@zkochan ok thanks, created an issue to track `pnpm pack --format json`

- https://github.com/pnpm/pnpm/issues/8764

**Comment by lvqq** [comment]
Refer PR: https://github.com/pnpm/pnpm/pull/8765

**Pull request merged**